lib/zclient.c

```diff
@@ -1465,6 +1465,47 @@ zebra_interface_vrf_update_read (struct stream *s, vrf_id_t vrf_id,
   *new_vrf_id = new_id;
   return ifp;
 }
+
+/* filter unwanted messages until the expected one arrives */
+static int
+zclient_read_sync_response (struct zclient *zclient, u_int16_t expected_cmd)
+{
+  struct stream *s;
+  u_int16_t size;
+  u_char marker;
+  u_char version;
+  vrf_id_t vrf_id;
+  u_int16_t cmd;
+  fd_set readfds;
+  int ret;
+
+  ret = 0;
+  cmd = expected_cmd + 1;
+  while (ret == 0 && cmd != expected_cmd)
```

Comment:
This doesn't work, we'll lose route updates that happen to be in the queue while a lm_get_label_chunk call is running.